### PR TITLE
[HW_LOCKS] replaced planned duration hours with deadline date

### PR DESCRIPTION
**Pull request by ranger1038**
Before:
Screenshot 2019-11-25 at 16 41 14
After:
Screenshot 2019-11-25 at 16 41 44

### CODE REVIEW

File: perftracker/static/js/perftracker.js [hunk 1]
```diff
@@ -65,6 +65,25 @@ function pt_dur2str(duration) {
     return rv + z(min) + ":" + z(sec);
 }
 
+function pt_date2ISOstr(date) {
+    date.setSeconds(0, 0);
```
Comment (istillc0de) [inline perftracker/static/js/perftracker.js]: method has unexpected side effects

File: perftracker/static/js/perftracker.js [hunk 2]
```diff
@@ -65,6 +65,25 @@ function pt_dur2str(duration) {
...
+    return date.toISOString().slice(0, -1);
+}
+
+function pt_getHoursBeforeDate(date) {
+    const now = new Date();
+    return Math.round(Math.abs(date - now) / (60 * 60 * 1000));
+}
+
+function pt_getDeadlineByDuration(dateFrom, duration) {
+    Date.prototype.addHours = function (hours) {
```
Comment (istillc0de) [inline perftracker/static/js/perftracker.js]: why modify prototype?!

File: perftracker/static/js/perftracker.js [hunk 3]
```diff
@@ -65,6 +65,25 @@ function pt_dur2str(duration) {
     return rv + z(min) + ":" + z(sec);
 }
 
+function pt_date2ISOstr(date) {
+    date.setSeconds(0, 0);
+    date = new Date(date.getTime() - date.getTimezoneOffset() * (60 * 1000));
+    return date.toISOString().slice(0, -1);
```
Comment (istillc0de) [inline perftracker/static/js/perftracker.js]: this method deserves good documentation and examples.
it is impossible to review it w/o running the code

File: perftracker/templates/_hw_lock_timeline.html [hunk 4]
```diff
@@ -92,21 +91,21 @@ <h4 class="modal-title" id="lockLabel"></h4>
...
     data.hw_nodes = [];
     $("#lockNodes").find('option:selected').each(function() {
         data.hw_nodes.push($(this).val());
     });
-    data.lock_type = {}
-    data.lock_type.name = $(".selected-lock-type").text();
-    data.lock_type.bg_color = $(".selected-lock-type").css('.background-color');
-    data.lock_type.fg_color = $(".selected-lock-type").css('.color');
+    data.lock_type = {
+        name: $('.selected-lock-type').text(),
```
Comment (istillc0de) [inline perftracker/templates/_hw_lock_timeline.html:103]: minor: consider to save JQuery element in variable

File: perftracker/static/js/perftracker.js (hunk 3 above)
Comment (ranger1038) [reply perftracker/static/js/perftracker.js]: https://stackoverflow.com/questions/28760254/assign-javascript-date-to-html5-datetime-local-input

File: perftracker/templates/_hw_lock_timeline.html (hunk 4 above)
Comment (ranger1038) [reply perftracker/templates/_hw_lock_timeline.html:103]: jQuery selectors are not stored in variables in the whole file, I've just followed existing codestyle

File: perftracker/static/js/perftracker.js (hunk 2 above)
Comment (ranger1038) [reply perftracker/static/js/perftracker.js]: https://stackoverflow.com/questions/1050720/adding-hours-to-javascript-date-object

File: perftracker/static/js/perftracker.js (hunk 1 above)
Comment (ranger1038) [reply perftracker/static/js/perftracker.js]: why? do we really need regulating seconds and ms while locking?

**Review by istillc0de — APPROVED** [review]

**Pull request merged**